### PR TITLE
Fix Bug 1136741, change wiki_spam_training flag to switch

### PR DESCRIPTION
**Pull request by schalkneethling**
Changes the `wiki_spam_training` flag to a switch

### CODE REVIEW

File: kuma/wiki/tests/test_forms.py [hunk 1]
```diff
@@ -537,8 +537,8 @@ def test_akismet_error(self, mock_requests):
     @pytest.mark.spam
     @requests_mock.mock()
     def test_akismet_spam_training(self, mock_requests):
-        flag, created = Flag.objects.get_or_create(name=SPAM_TRAINING_FLAG)
-        flag.users.add(self.testuser)
+        switch, created = Switch.objects.get_or_create(name=SPAM_TRAINING_SWITCH)
+        switch.users.add(self.testuser)
```
Comment (schalkneethling) [inline kuma/wiki/tests/test_forms.py]: @jwhitlock Not sure how to handle this as the `Switch` object does not have a user property.

Comment (schalkneethling) [reply kuma/wiki/tests/test_forms.py]: So, from the code I see that "Switches are active, or inactive, globally." so, perhaps I should just set active to `True` and remove line 541?

Comment (jwhitlock) [reply kuma/wiki/tests/test_forms.py]: This code is being overly cautious, and can be replaced with:

```
Switch.objects.create(name=SPAM_TRAINING_SWITCH, active=True)
```

Same for ``test_akismet_error_training`` below.

**Comment by schalkneethling** [comment]
Thanks for the review @jwhitlock - updated.

**Comment by jwhitlock** [comment]
It looks like the rebase wasn't clean - making some small changes.

**Pull request merged**